packages/eslint-plugin/src/rules/no-unnecessary-parameter-property-assignment.ts

```diff
@@ -0,0 +1,241 @@
+import { DefinitionType } from '@typescript-eslint/scope-manager';
+import type { TSESTree } from '@typescript-eslint/utils';
+import { AST_NODE_TYPES, ASTUtils } from '@typescript-eslint/utils';
+
+import { createRule, getStaticStringValue, nullThrows } from '../util';
+
+type MessageIds = 'unnecessaryAssign';
+
+const UNNECESSARY_OPERATORS = new Set(['=', '&&=', '||=', '??=']);
+
+export default createRule<[], MessageIds>({
+  name: 'no-unnecessary-parameter-property-assignment',
+  meta: {
+    docs: {
+      description:
+        'Disallow unnecessary assignment of constructor property parameter',
+      requiresTypeChecking: false,
+    },
+    fixable: 'code',
+    messages: {
+      unnecessaryAssign:
+        'This assignment is unnecessary since it already assigned by parameter property.',
+    },
+    schema: [],
+    type: 'suggestion',
+  },
+  defaultOptions: [],
+  create(context) {
+    const reportInfoStack: {
+      assignedBeforeUnnecessary: Set<string>;
+      assignedBeforeCtor: Set<string>;
```

Comment:
[Naming] Nit: I always have a hard time remembering abbreviations. Up for a little more verbosity?

```suggestion
      assignedBeforeConstructor: Set<string>;
```